app/queries/my_facilities/missed_visits_query.rb

```diff
@@ -0,0 +1,86 @@
+class MyFacilities::MissedVisitsQuery
+  # Wrap query method calls with the appropriate timezone in which the reports will be consumed
+  # This is probably the ENV['ANALYTICS_TIME_ZONE']
+  # Example: `Time.use_zone('timezone string') { bp_control_query_object.cohort_registrations }`
+
+  include QuarterHelper
+  include MonthHelper
+
+  attr_reader :periods
+
+  def initialize(facilities: Facility.all, period: :quarter, last_n: 3)
+    # period can be :quarter, :month.
+    # last_n is the number of quarters/months data to be returned
+    @facilities = facilities
+    @period = period
+    @periods = period_list(period, last_n)
+    @latest_period = @periods.first
+  end
+
+  def patients
+    @patients ||=
+      Patient
```

Comment:
Hmm, I think we can leave it as-is. The latest_bp from the views will be updated only once a day, so it can be stale. 
The view also returns a LatestBloodPressurePerPatient object instead of a BloodPressure object, this might not be what we want from Patient.latest_blood_pressures. We'll have to join this with the BloodPressures table to return a BP object

Reply:
We should try to have a single source of truth for all our definitions. If we are using the view in our reports, we should use it else where in the dashboard as well.